bindings/java/src/org/sleuthkit/datamodel/SleuthkitJNI.java

```diff
@@ -228,19 +233,19 @@ private AddImageProcess(String timeZone, boolean addUnallocSpace, boolean skipFa
 			 *                          the process)
 			 */
 			public void run(String deviceId, String[] imageFilePaths) throws TskCoreException, TskDataException {
-				if (0 != tskAutoDbPointer) {
-					throw new TskCoreException("Add image process already started");
-				}
-
-				long imageHandle = openImage(imageFilePaths, false);
-				
-				synchronized (this) {					
-					tskAutoDbPointer = initAddImgNat(caseDbPointer, timezoneLongToShort(timeZone), addUnallocSpace, skipFatFsOrphans);
-				}
-				if (0 == tskAutoDbPointer) {
-					throw new TskCoreException("initAddImgNat returned a NULL TskAutoDb pointer");
+				long imageHandle;
+				synchronized (this) {
+					if (0 != tskAutoDbPointer) {
+						throw new TskCoreException("Add image process already started");
+					}
+					imageHandle = openImage(imageFilePaths, false);
```

Comment:
It won't hurt anything to have another check, but it would be nearly impossible for a user to hit cancel before we get to this point.